src/index.tsx

```diff
@@ -18,32 +23,9 @@ function createResizedImage(
   rotation?: number,
   outputPath?: string,
   keepMeta?: boolean,
-  options: {
-    /**
-     * Either `contain` (the default), `cover`, or `stretch`. Similar to
-     * [react-native <Image>'s resizeMode](https://reactnative.dev/docs/image#resizemode)
-     *
-     * - `contain` will fit the image within `width` and `height`,
-     *   preserving its ratio
-     * - `cover` will make sure at least one dimension fits `width` or
-     *   `height`, and the other is larger, also preserving its ratio.
-     * - `stretch` will resize the image to exactly `width` and `height`.
-     *
-     * (Default: 'contain')
-     */
-    mode?: ResizeMode;
-    /**
-     * Whether to avoid resizing the image to be larger than the original.
-     * (Default: false)
-     */
-    onlyScaleDown?: boolean;
-  } = {
-    mode: 'contain',
-    onlyScaleDown: false,
-  }
+  options: Options = defaultOptions
 ): Promise<Response> {
-  const mode = options.mode;
-  const onlyScaleDown = options.onlyScaleDown;
+  const { mode, onlyScaleDown } = { ...defaultOptions, ...options };
```

Comment:
In the previous version, we had a method that validate mode and onlyScaleDown values, I would keep this validation + error throwing because if some users rely on this error being thrown, their implementation is now broken

Reply:
``` /** Validate `options` object: used by both Android and iOS entry points */
function validateOptions(options) {
  const mode = options.mode || 'contain';
  const possibleModes = ['contain', 'cover', 'stretch'];
  if (possibleModes.indexOf(mode) === -1) {
    throw new Error(`createResizedImage's options.mode must be one of "${possibleModes.join('", "')}"`);
  }

  if (options.onlyScaleDown && typeof options.onlyScaleDown !== 'boolean') {
    throw new Error(`createResizedImage\'s option.onlyScaleDown must be a boolean: got ${options.onlyScaleDown}`);
  }

  return {
    mode,
    onlyScaleDown: !!options.onlyScaleDown,
  };
}```